pkg/instancetype/compatibility.go

```diff
@@ -11,18 +11,29 @@ import (
 	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
 )
 
-func decodeOldObject(data []byte, isPreference bool) (runtime.Object, error) {
-	if isPreference {
-		oldObject, err := decodeOldPreferenceRevisionObject(data)
-		if err != nil {
-			return nil, fmt.Errorf("failed to decode old revision object: %w", err)
-		}
-		if oldObject == nil {
-			return nil, nil
-		}
-		return convertPreference(oldObject)
+func isInstancetypeSpec(data []byte) bool {
+	return json.Unmarshal(data, &instancetypev1alpha1.VirtualMachineInstancetypeSpec{}) == nil
+}
+
+func isPreferenceSpec(data []byte) bool {
+	return json.Unmarshal(data, &instancetypev1alpha1.VirtualMachinePreferenceSpec{}) == nil
+}
+
+func decodeOldObject(data []byte) (runtime.Object, error) {
+	if isInstancetypeSpec(data) {
+		return decodeOldInstancetypeObject(data)
 	}
+	if isPreferenceSpec(data) {
+		return decodeOldPreferenceObject(data)
+	}
+	var obj runtime.Object
+	if err := json.Unmarshal(data, obj); err != nil {
+		return nil, err
+	}
```

Comment:
This `json.Unmarshal()` call will fail, because `obj` is `nil`, so it cannot fill it with values. The call uses reflection on the actual struct passed as second parameter, but it this case the parameter is `nil`, so it does not know how to create it.

Reply:
As above, made an assumption about coverage that turned out to be completely bogus, apologies for wasting your time with that revision of the PR.